Propose new API endpoint

smithjac opened the pull request:
Within the API documentations, a new API endpoint
is specified. It is called 'search' and takes a
term that the caller wants to search for.

The new API endpoint will use all three of the
previous endpoints for the search.

bjcoleman commented:
Not sure why you get a conflict here - possibly you didn't pull from `upstream main` first.  But if you pull into your branch (be in `pylint` and run `git pull upstream main`) you can resolve the conflict and then commit/push the merged version.

MarieSAddo commented:
looks good. will merge

Commit:
Propose new API endpoint

Within the API documentations, a new API endpoint
is specified. It is called 'search' and takes a
term that the caller wants to search for.

The new API endpoint will use all three of the
previous endpoints for the search.

Commit:
Implement basic search endpoint

Began implementing the search endpoint for the
API. Currently it has no funtionality, it just
checks if there is a search term provided.

Commit:
Address requested changes on PR

Bolded the phrase "case sensitive" to make sure
it is made known.

tabato commented on docs/API.md:
```diff
@@ -1,8 +1,57 @@
+
+### Calling the Endpoint
+`/search?term=SEARCH_PARAMETER`
+To call this endpoint, make a request using the example above. When replacing the `term` to be searched, it is important to point out that all search terms are case sensitive. 
```
Let's try and add some highlight to the fact that these are CASE SENSITIVE. It's a small thing that can cause big issues if missed.

tabato commented on docs/API.md:
````diff
@@ -1,8 +1,57 @@
+### Endpoint Return
+* `200`<br>
+```
+{
````
Having this as code makes sense, but if we can add a way to make it more digestible before or after would be awesome.
